Cura.proto

```diff
@@ -61,6 +61,23 @@ message Polygon {
     float line_width = 3; // The width of the line being laid down
 }
 
+message LayerOptimized {
+    int32 id = 1;
+    float height = 2; // Z position
+    float thickness = 3; // height of a single layer
+
+    repeated PathSegment path_segment = 4; // layer data
```

Comment:
One idea when this is beneficial, related to your earlier comment, is when using multiple extruders. Perhaps a new field should be added to the PathSegment message to indicate which extruder is active. Then you would get as many PathSegments as you have extruders. Could of course be implemented in another way using multiple layers or line type offsets.